.circleci/config.yml

```diff
@@ -1,5 +1,23 @@
 version: 2.0
+
+flake8-steps: &steps
+  - checkout
+  - run: pip install --user flake8
+  - run: python -m flake8 --version ; pip --version
+  # stop the build if there are Python syntax errors or undefined names
+  - run: python -m flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics
```

Comment:
I really do not like separate configuration files.  They hide what you are trying to do in one of three separate places so that when I run flake8 in one project, it does something different than it does in another project.  The other problem is that this PR is running flake8 with two different configurations.  We lose that kind of flexibility if we lock our flake8 preferences in some separate configuration file.

An alternative that I could recommend that we do is create a separate __flake8.tests.sh__ that contains both flake8 runs and we call that script from __.circleci/config.yml __ and encourage contributors to do the same locally.

Reply:
@cclauss's logic here makes sense.

Reply:
Sure, a standalone script fits that bill as well. My worries are only about tightly coupling CI configuration with functionality that is also useful during development.